integrations/tensorflow/e2e/keras/layers/layers_test.py

```diff
@@ -0,0 +1,270 @@
+    'Add': [RANK_2_INPUT, RANK_2_INPUT],
+    'AdditiveAttention': [RANK_3_INPUT, RANK_3_INPUT, RANK_3_INPUT],
+    'AlphaDropout': [RANK_2_INPUT],
+    'Attention': [RANK_3_INPUT, RANK_3_INPUT, RANK_3_INPUT],
+    'Average': [RANK_2_INPUT, RANK_2_INPUT],
+    'AveragePooling1D': [CONV_1D_INPUT],
+    'AveragePooling2D': [CONV_2D_INPUT],
+    'AveragePooling3D': [CONV_3D_INPUT],
+    'BatchNormalization': [RANK_2_INPUT],
+    'Concatenate': [RANK_4_INPUT, RANK_4_INPUT],
+    'Conv1D': [CONV_1D_INPUT],
+    'Conv1DTranspose': [CONV_1D_INPUT],
+    'Conv2D': [CONV_2D_INPUT],
+    'Conv2DTranspose': [CONV_2D_INPUT],
+    'Conv3D': [CONV_3D_INPUT],
+    'Conv3DTranspose': [CONV_3D_INPUT],
+    'ConvLSTM2D': [CONV_3D_INPUT],
+    'Cropping1D': [CONV_1D_INPUT],
+    'Cropping2D': [CONV_2D_INPUT],
+    'Cropping3D': [CONV_3D_INPUT],
+    'Dense': [RANK_2_INPUT],
+    'DepthwiseConv2D': [CONV_2D_INPUT],
+    'Dot': [RANK_3_INPUT, RANK_3_INPUT],
+    'Dropout': [RANK_3_INPUT],
+    'ELU': [RANK_2_INPUT],
+    'Embedding': [RANK_2_INPUT],
+    'Flatten': [RANK_2_INPUT],
+    'GRU': [RANK_3_INPUT],
+    'GRUCell': [RANK_2_INPUT, RANK_2_INPUT],
+    'GaussianDropout': [RANK_2_INPUT],
+    'GaussianNoise': [RANK_2_INPUT],
+    'GlobalAveragePooling1D': [CONV_1D_INPUT],
+    'GlobalAveragePooling2D': [CONV_2D_INPUT],
+    'GlobalAveragePooling3D': [CONV_3D_INPUT],
+    'GlobalMaxPool1D': [CONV_1D_INPUT],
+    'GlobalMaxPool2D': [CONV_2D_INPUT],
+    'GlobalMaxPool3D': [CONV_3D_INPUT],
+    'InputLayer': [RANK_2_INPUT],
+    'LSTM': [RANK_3_INPUT],
+    'LSTMCell': [RANK_2_INPUT, RANK_2_INPUT],
+    'Lambda': [RANK_2_INPUT],
+    'LayerNormalization': [RANK_2_INPUT],
+    'LeakyReLU': [RANK_2_INPUT],
+    'LocallyConnected1D': [CONV_1D_INPUT],
+    'LocallyConnected2D': [CONV_2D_INPUT],
+    'Masking': [RANK_2_INPUT],
+    'MaxPool1D': [CONV_1D_INPUT],
+    'MaxPool2D': [CONV_2D_INPUT],
+    'MaxPool3D': [CONV_3D_INPUT],
+    'Maximum': [RANK_2_INPUT, RANK_2_INPUT],
+    'Minimum': [RANK_2_INPUT, RANK_2_INPUT],
+    'MultiHeadAttention': [RANK_3_INPUT, RANK_3_INPUT],
+    'Multiply': [RANK_2_INPUT, RANK_2_INPUT],
+    'PReLU': [RANK_2_INPUT],
+    'Permute': [RANK_4_INPUT],
+    'ReLU': [RANK_2_INPUT],
+    'RepeatVector': [RANK_2_INPUT],
+    'Reshape': [RANK_3_INPUT],
+    'SeparableConv1D': [CONV_1D_INPUT],
+    'SeparableConv2D': [CONV_2D_INPUT],
+    'SimpleRNN': [RANK_3_INPUT],
+    'SimpleRNNCell': [RANK_2_INPUT, RANK_2_INPUT],
+    'Softmax': [RANK_2_INPUT],
+    'SpatialDropout1D': [CONV_1D_INPUT],
+    'SpatialDropout2D': [CONV_2D_INPUT],
+    'SpatialDropout3D': [CONV_3D_INPUT],
+    'Subtract': [RANK_2_INPUT, RANK_2_INPUT],
+    'ThresholdedReLU': [RANK_2_INPUT],
+    'UpSampling1D': [CONV_1D_INPUT],
+    'UpSampling2D': [CONV_2D_INPUT],
+    'UpSampling3D': [CONV_3D_INPUT],
+    'ZeroPadding1D': [CONV_1D_INPUT],
+    'ZeroPadding2D': [CONV_2D_INPUT],
+    'ZeroPadding3D': [CONV_3D_INPUT],
+}
+
+LAYER_TO_KWARGS = {
```

Comment:
Maybe a default_dict to avoid having to list all these ones that just take an empty dict again?

Reply:
I decided to use `dict()` to explicitly document that we weren't modifying any of the layer's settings. It is pretty verbose though. Thoughts?